modular_skyrat/master_files/code/modules/client/preferences/_preference.dm

```diff
@@ -75,3 +75,88 @@
+
+/datum/preference/toggle/mutant_toggle/is_accessible(datum/preferences/preferences)
+	var/passed_initial_check = ..(preferences)
+	var/allowed = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
+	return passed_initial_check || allowed
+
+/**
+ * Base class for choices character features, mainly mutant body parts
+ */
+/datum/preference/choiced/mutant_choice
+	abstract_type = /datum/preference/choiced/mutant_choice
+	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+	savefile_identifier = PREFERENCE_CHARACTER
+
+	/// Path to the default sprite accessory
+	var/datum/sprite_accessory/default_accessory_type
+	/// Path to the corresponding /datum/preference/toggle to check if part is enabled.
+	var/datum/preference/toggle/type_to_check
+
+/datum/preference/choiced/mutant_choice/is_accessible(datum/preferences/preferences)
+	var/passed_initial_check = ..(preferences)
+	var/overriding = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
+	var/part_enabled = is_part_enabled(preferences)
+	return (passed_initial_check || overriding) && part_enabled
+
+/datum/preference/choiced/mutant_choice/init_possible_values()
+	return assoc_to_keys(GLOB.sprite_accessories[relevant_mutant_bodypart])
+
+/datum/preference/choiced/mutant_choice/create_default_value()
+	return initial(default_accessory_type?.name) || "None"
+
+/**
+ * Is this part enabled by the player?
+ *
+ * Arguments:
+ * * preferences - The relevant character preferences.
+ */
+/datum/preference/choiced/mutant_choice/proc/is_part_enabled(datum/preferences/preferences)
+	return preferences.read_preference(type_to_check)
+
+/**
+ * Actually rendered. Slimmed down version of the logic in is_available() that actually works when spawning or drawing the character.
+ *
+ * Returns if feature is visible.
+ *
+ * Arguments:
+ * * The character this is being applied to.
+ * * preferences - The relevant character preferences.
+ */
+/datum/preference/choiced/mutant_choice/proc/is_visible(mob/living/carbon/human/target, datum/preferences/preferences)
+	var/species_type = preferences.read_preference(/datum/preference/choiced/species)
+	var/datum/species/species = new species_type
+
+	var/species_available = (savefile_key in species.get_features())
+
+	var/overriding = preferences.read_preference(/datum/preference/toggle/allow_mismatched_parts)
+	var/part_enabled = is_part_enabled(preferences)
+	return (species_available || overriding) && part_enabled
+
+/datum/preference/choiced/mutant_choice/apply_to_human(mob/living/carbon/human/target, value, datum/preferences/preferences)
+	if(!preferences || !is_visible(target, preferences))
+		value = create_default_value()
+		. = FALSE
+
+	if(!target.dna.mutant_bodyparts[relevant_mutant_bodypart])
+		target.dna.mutant_bodyparts[relevant_mutant_bodypart] = list(MUTANT_INDEX_NAME = value, MUTANT_INDEX_COLOR_LIST = list("#FFFFFF", "#FFFFFF", "#FFFFFF"), MUTANT_INDEX_EMISSIVE_LIST = list(FALSE, FALSE, FALSE))
```

Comment:
Couldn't we use the default `Mutant Colors` here rather than just going to white? Just a thought, would make a bit more sense to me personally, but it's really whatever.

Reply:
The code this was replacing is white - I'm guessing it doesn't matter.